### ISSUE TITLE
Null values being returned when referencing dictionary

### ISSUE THREAD

**Issue by MrMemeikyu** [MrMemeikyu/Heist-Earpiece#1]
Null output:
<img width="567" alt="Screenshot 2024-11-11 at 12 29 13 PM" src="https://github.com/user-attachments/assets/75507171-62a6-44a1-82cc-048e11ec6d3a">
<img width="735" alt="Screenshot 2024-11-11 at 12 29 24 PM" src="https://github.com/user-attachments/assets/640559af-a206-46e1-a4ed-a6e895171cb9">

Code producing null outputs (when referring to dictionary sort):
<img width="602" alt="Screenshot 2024-11-11 at 12 43 45 PM" src="https://github.com/user-attachments/assets/180d1564-d22e-46a1-a633-d07877b147b0">

I have tried to add an if statement to remove said null value, but either it isn't working or is being changed back to a null. I've also messed around with the values getting turned into null values and they never seem to line up in any reasonable way.

**Comment by MrMemeikyu** [comment]
Only happens with the nature category. The key and value match the input, but for some reason the function is still returning as null from the value in the dictionary

**Comment by MrMemeikyu** [comment]
Also happens with colour category, same thing happens when I change the name too, but for the time being I'm just cutting the two out completely and it seems to work from there. Temporary solution that works and may be a pain to fix properly later

**Comment by paralinguist** [comment]
Had a look and it was hard to follow exactly how the selection process was working which makes it tricky to identify null generation.

I've rewritten the selection process to be (hopefully) simpler - take a look at the code and see if it makes sense to you.

**Comment by MrMemeikyu** [comment]
THANK YOU SIR that rework is much more efficient and coherent appreciate rewriting all that
